### PR TITLE
Add format test to CI

### PR DESCRIPTION
**Pull request by costinsin**
To enforce uniformity for all PRs, the CI checks if the code
is formatted right using `cargo fmt` tool.

Results after implementing the format test in CicleCI, but before fixing the format errors: https://cirrus-ci.com/build/4684991404703744
Results after fixing the format errors: https://cirrus-ci.com/build/5423803479097344

Solves #770

### CODE REVIEW

File: .cirrus.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ build: &BUILD
     - . $HOME/.cargo/env || true
     - $TOOL +$TOOLCHAIN -Vv
     - rustc +$TOOLCHAIN -Vv
+    - if [ -z "$TOOLCHAIN" ]; then rustup component add rustfmt; else rustup +$TOOLCHAIN component add rustfmt; fi
+    - $TOOL +$TOOLCHAIN fmt --all -- --check
```
Comment (asomers) [inline .cirrus.yml:22]: `cargo fmt` ignores cfg directives, so running it once suffices to provide coverage for every OS.  Running it on every OS is duplicative.  Instead, you should either create a new task just for this, or else put it into the "Rust Stable" task.

**Review by asomers — CHANGES_REQUESTED** [review]
Call me old-fashioned, but I still prefer 80-characters lines.  Could you please add a rustfmt.toml file that contains:
```
max_width = 80
```

File: .cirrus.yml (hunk 1 above)
Comment (costinsin) [reply .cirrus.yml]: I have created a new CI task. This way, if this is the only test failing, the person who submitted the PR will easily know that the PR failed due to bad formatting. I have tested the commit without applying the formatting changes and [everything seems fine](https://cirrus-ci.com/build/4684991404703744)

Comment (costinsin) [reply .cirrus.yml]: I have also added the `rustfmt.toml` file specifying that he max width needs to be 80 characters.

File: .cirrus.yml [hunk 2]
```diff
@@ -320,3 +320,11 @@ task:
   check_script:
     - cargo check
   before_cache_script: rm -rf $CARGO_HOME/registry/index
+
+# Tasks that checks if the code is formatted right using `cargo fmt` tool
+task:
+  name: Rust Formatter
```
[the remark this answers is not in the document]
Comment (costinsin) [reply .cirrus.yml:326]: Solved!

**Review by rtzoeller — APPROVED** [review]
bors r+

**Pull request merged**